src/NUnitFramework/framework/Internal/Execution/ParallelWorkItemDispatcher.cs

```diff
@@ -65,17 +65,17 @@ private void InitializeShifts()
             // TODO: Avoid creating all the workers till needed
             for (int i = 1; i <= LevelOfParallelism; i++)
             {
-                string name = $"ParallelWorker#{i}";
+                string name = $"NUnit.Fw.ParallelWorker#{i}";
                 ParallelShift.Assign(new TestWorker(ParallelQueue, name));
             }
 
-            ParallelShift.Assign(new TestWorker(ParallelSTAQueue, "ParallelSTAWorker"));
+            ParallelShift.Assign(new TestWorker(ParallelSTAQueue, "NUnit.Fw.ParallelSTAWorker"));
 
-            var worker = new TestWorker(NonParallelQueue, "NonParallelWorker");
+            var worker = new TestWorker(NonParallelQueue, "NUnit.Fw.NonParallelWorker");
             worker.Busy += OnStartNonParallelWorkItem;
             NonParallelShift.Assign(worker);
 
-            worker = new TestWorker(NonParallelSTAQueue, "NonParallelSTAWorker");
+            worker = new TestWorker(NonParallelSTAQueue, "NUnit.Fw.NonParallelSTAWorker");
```

Comment:
What do you think about leaving the NUnit names for the workers the same, but prepending `NUnit.Fw` at the point of assigning `Thread.Name`?

Reply:
Sure , that could also work. 
We're then assuming that TestWorker is not called from the outside, as it is a public method, even if it is in the Internal namespace.

Reply:
Just thinking, since `NUnit.Fw` was motivated by being able to distinguish threads, perhaps it "belongs" with Thread.Name and not in logs. But I don't feel strongly

Reply:
Agree there.  I actually don't like that we're sending names of something around the way we do here and in some other places.  If the name should be close to where the thread are created, it should be in that parent class.  To me it looks like there should be different classes for each of these, and not a single class like now.
If we had a 
public class NonParallelTestWorker : TestWorker which handled the initialization, including name, this would resolve easily:
The current code is:
`var worker = new TestWorker(NonParallelQueue, "NUnit.Fw.NonParallelWorker");
worker.Busy += OnStartNonParallelWorkItem;
NonParallelShift.Assign(worker);`
could all be moved into NonParallelTestWorker .  Same with the OnStartNonParallelWorkItem, it only works on the TestWorker and the Workitem.

Reply:
I'd personally think of disassociating the class/log name from the thread name, since the thread names are more to look "pretty" for easy understanding when debugging. Though come to think of it, perhaps folks will want to correlate it with the log.

I'm good with your approach!